helpers/deproxy.py

```diff
@@ -882,13 +882,13 @@ def __init__(self, request, expected_response, forwarded_request=None,
         # Response received on the client.
         self.response = expected_response
         # Expected request forwarded by Tempesta to the server.
-        self.fwd_request = forwarded_request if forwarded_request else Request()
+        self.fwd_request = forwarded_request if forwarded_request else None
```

Comment:
It took me some time to identify why `Request()` was used here as a default value before the patch. And it turned down that `helpers.Deproxy` and `helpers.chains` are inconsistent: while one uses empty class instance for absent message, the other uses `None`. The first one was historically first, and was used to avoid null pointer dereferences. The second one was used later (once `chains` was introdused), and  `None` was a marker that there is no message. 

I also see, that we already faced troubles from that inconsistency. Some functions are wise enough to support both `None` and `Request()` in the same time, e.g. `helpers.deproxy.Deproxy.check_expectations()`. But some test cases actually depends on  expected format. I really appreciate you pointing out this issue. And I agree with you that the most recent change (with `None` as default) should be chosen as the only solution.

The issue is very old though and it's difficult to say how many tests are affected. If you run the whole test suite and have no failures due `Request()/None` misuse in elder tests - then merge the PR. But if a huge bunch of troubles was introduced, then I believe we should drop the change for now, report it and schedule for 0.8.

Reply:
Oh, forgot to mention:
```
self.fwd_request = forwarded_request if forwarded_request else None
```
semantically equals to 
```
self.fwd_request = forwarded_request
```